webclient/src/components/CalendarModal.vue

```diff
@@ -0,0 +1,170 @@
+<script setup lang="ts">
+import { CalendarView, CalendarViewHeader } from "vue-simple-calendar";
+import { ref, computed, watch } from "vue";
+import type { ICalendarItem } from "vue-simple-calendar/dist/src/ICalendarItem";
+
+import { useGlobalStore } from "@/stores/global";
+import { useRoute } from "vue-router";
+import type { components } from "@/api_types";
+type CalendarResponse = components["schemas"]["CalendarResponse"];
+import "/node_modules/vue-simple-calendar/dist/style.css";
+import "/node_modules/vue-simple-calendar/dist/css/gcal.css";
+import { useFetch } from "@/composables/fetch";
+
+const global = useGlobalStore();
+const showDate = ref(new Date());
+
+const tumonlineCalendarUrl = ref("https://campus.tum.de/tumonline");
+const last_sync = ref("xx.xx.xxxx xx:xx");
+const events = ref<ICalendarItem[]>([]);
+
+const route = useRoute();
+
+const start = computed(() => {
+  const start = new Date(showDate.value);
+  start.setDate(start.getDate() - 60);
+  return start.toISOString().replace("Z", "");
+});
+
+const end = computed(() => {
+  const start = new Date(showDate.value);
+  start.setDate(start.getDate() + 60);
+  return start.toISOString().replace("Z", "");
+});
+// called when the view is loaded
+update();
+// called when the view navigates to another view, but not when its initially loaded
+watch(() => showDate.value, update);
+watch(() => route.params.id, update);
+
+function update() {
```

Comment:
Consider using a more descriptive name for the 'update' function, such as 'updateCalendarEvents', to better reflect its purpose. [medium]